addons/zookeeper/scripts/startup.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Execute entrypoint as usual after obtaining ZOO_SERVER_ID
+# check ZOO_SERVER_ID in persistent volume via myid
+# if not present, set based on POD hostname
+# Execute entrypoint as usual after obtaining ZOO_SERVER_ID
+# check ZOO_SERVER_ID in persistent volume via myid
+# if not present, set based on POD hostname
+if [[ -f "/bitnami/zookeeper/data/myid" ]]; then
+  export ZOO_SERVER_ID="$(cat /bitnami/zookeeper/data/myid)"
+else
+  SERVICE_ID=$((${KB_POD_NAME##*-}))
+  export ZOO_SERVER_ID="$((SERVICE_ID))"
+  echo $ZOO_SERVER_ID > /bitnami/zookeeper/data/myid
+fi
+
+function version_gt() { test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" != "$1"; }
+function version_le() { test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" == "$1"; }
+function version_lt() { test "$(echo "$@" | tr " " "\n" | sort -rV | head -n 1)" != "$1"; }
+function version_ge() { test "$(echo "$@" | tr " " "\n" | sort -rV | head -n 1)" == "$1"; }
+
+
+if  version_gt "3.6.0" "${ZOOKEEPER_IMAGE_VERSION%%-*}"  ; then
```

Comment:
Consider the scenario where ZOOKEEPER_IMAGE_VERSION is empty. This env variable doesn't exist in legacy cluster.